pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -70,23 +74,22 @@ func (k *Kaniko) BuildAndPushContainerImage(buildOptions *BuildOptions, namespac
 
 		if runningJob.Status.Succeeded > 0 {
 			k.logger.Debug("Kaniko job was completed successfully")
-
-			// Cleanup
-			propagationPolicy := meta_v1.DeletePropagationBackground
-			err = k.kubeClientSet.BatchV1().Jobs(namespace).Delete(kanikoJob.Name, &meta_v1.DeleteOptions{
-				PropagationPolicy: &propagationPolicy,
-			})
-			if err != nil {
-				k.logger.Error("Failed to delete Kaniko job after successful completion")
-			}
 			return nil
 		}
 		if runningJob.Status.Failed > 0 {
-			k.logger.DebugWith("Kaniko job has failed", "status", runningJob.Status)
-			return errors.New("Kaniko job has failed")
+			k.logger.Warn("Kaniko job has failed", "status", runningJob.Status)
```

Comment:
aggregate into one line of "Kaniko job has failed" (it will be a bit confusing reading duplicate log line)

Reply:
A Log + return err should rarely happen. If you do this, it means you don't log the other errors that happen in your code.
You should always have one or two places in your code that log errors that happen (e.g. where you start() your app and where you handle requests).